client/updates/yum.go

```
@@ -0,0 +1,118 @@
+	if err != nil {
+		return nil, err
+	}
+
+	securityUpdates, err := p.listUpdates(ctx, "--security")
+	if err != nil {
+		return nil, err
+	}
+	isSecurity := make(map[string]bool, len(securityUpdates))
+	for _, u := range securityUpdates {
+		isSecurity[u.name] = true
+	}
+
+	summaries := make([]models.UpdateSummary, len(allUpdates))
+	for i, update := range allUpdates {
+		summaries[i] = models.UpdateSummary{
+			Title:            update.name,
+			Description:      fmt.Sprintf("%s %s [%s] (%s)", update.name, update.version, update.arch, update.repository),
+			IsSecurityUpdate: isSecurity[update.name],
+		}
+	}
+
+	rebootPending := p.checkRebootRequired(ctx)
+
+	return &models.UpdatesStatus{
+		UpdatesAvailable:         len(allUpdates),
+		SecurityUpdatesAvailable: len(securityUpdates),
+		UpdateSummaries:          summaries,
+		RebootPending:            rebootPending,
+	}, nil
+}
+
+type yumUpdate struct {
+	name       string
+	arch       string
+	version    string
+	repository string
+}
+
+func (p *YumPackageManager) listUpdates(ctx context.Context, additionalArgs ...string) ([]yumUpdate, error) {
+	cmd := append([]string{"check-update", "--quiet"}, additionalArgs...)
+	output, err := p.run(ctx, cmd...)
+	// exit code 100 means updates available, so continue
+	if err != nil && err.Error() != "exit status 100" {
+		return nil, err
+	}
+
+	result := []yumUpdate{}
+	for _, line := range strings.Split(output, "\n") {
+		parts := strings.Fields(line)
+		if len(parts) != 3 {
+			continue
+		}
+
+		update := yumUpdate{
+			name:       parts[0],
+			version:    parts[1],
+			repository: parts[2],
+		}
+
+		nameArch := strings.Split(parts[0], ".")
+		if len(nameArch) == 2 {
+			update.name = nameArch[0]
+			update.arch = nameArch[1]
+		}
+
+		result = append(result, update)
+	}
+
+	return result, nil
+}
+
+func (p *YumPackageManager) checkRebootRequired(ctx context.Context) bool {
+	output, _ := p.run(ctx, "needs-restarting", "-r")
```

Would be good not to silence the error, return it and handle it higher.

---

Please add an explanatory comment on why the err is discarded.